psydac/feec/global_projectors.py

```diff
@@ -0,0 +1,472 @@
+# -*- coding: UTF-8 -*-
+
+from psydac.linalg.utilities      import array_to_stencil
+from psydac.linalg.kron           import kronecker_solve
+from psydac.linalg.stencil        import StencilVector
+from psydac.linalg.block          import BlockVector
+from psydac.core.bsplines         import quadrature_grid
+from psydac.utilities.quadratures import gauss_legendre
+from psydac.fem.basic             import FemField
+from psydac.fem.vector            import VectorFemField
+
+class Projector_H1:
+
+    def __init__(self, H1):
+
+        # Quadrature grids in cells defined by consecutive Greville points
+
+        points = [V.greville for V in H1.spaces]
+
+        H1.init_interpolation()
+
+        # Collocation matrices for N-splines in each direction
+        self.N   = [V._interpolator for V in H1.spaces]
+        n_basis  = [V.nbasis for V in H1.spaces]
+        self.rhs = StencilVector(H1.vector_space)
+        slices   = tuple(slice(p,-p) for p in H1.degree)
+
+        self.space = H1
+        self.args  = (*n_basis, *points, self.rhs._data[slices])
+
+        if len(self.N) == 1:
+            self.func = evaluate_dof_0form_1d
+        elif len(self.N) == 2:
+            self.func = evaluate_dof_0form_2d
+        elif len(self.N) == 3:
+            self.func = evaluate_dof_0form_3d
+        else:
+            raise ValueError('H1 projector of dimension {} not available'.format(str(len(self.N))))
+
+    # ======================================
+    def __call__(self, fun):
+        r'''
+        Projection on the space V0 via interpolation.
+
+        Parameters
+        ----------
+        fun : callable
+            fun(x) \in R is the 0-form to be projected.
+
+        Returns
+        -------
+        coeffs : 1D array_like
+            Finite element coefficients obtained by projection.
+        '''
+
+        # build the rhs
+        self.func(*self.args, fun)
+
+        if len(self.N)==1:
+            rhs = self.rhs.toarray()
+            coeffs = array_to_stencil(self.N[0].solve(rhs), self.space.vector_space)
+            coeffs.update_ghost_regions()
+        else:
+            coeffs = kronecker_solve(solvers = self.N, rhs = self.rhs)
```

Comment:
As you can see [here](https://github.com/pyccel/psydac/blob/fb5f9d6552ced9ccc3d552f2fa6f8096bd70d6a8/psydac/linalg/kron.py#L417) this function accepts an `out` keyword argument. Hence this line should read
```python
kronecker_solve(solvers = self.N, rhs = self.rhs, out=coeffs)
```